client/lib/user-settings/index.js

```diff
@@ -29,6 +26,26 @@ function decodeUserSettingsEntities( data ) {
 	return assign( {}, data, decodedValues );
 }
 
+/*
+ * Deletes a provided unsaved setting, then calls itself recursively
+ * to delete any empty parents of the setting passed to it
+ */
+function deleteUnsavedSetting( settings, settingName, recursive ) {
+	if ( ! isEmpty( get( settings, settingName ) ) || recursive ) {
+		unset( settings, settingName );
+
+		const settingKeys = settingName.split( '.' );
+		if ( settingKeys.length > 1 ) {
```

Comment:
Hmm, `! isEmpty( settingKeys )` would be true though even if there is only one setting key. We only want to continue if there is more than one key e.g. `["calypso_settings", "colorScheme"]`.